### PR TITLE
Refine the implementation of fopen/fdopen

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- stdio: Move lib_stream_[initialize|release] from libc/stdio to sched/tls 
- stdio: Initialize stdin, stdout and stderr directly
- stdio: Merge fs_fdopen into fdopen to simplify the code logic

## Impact

stdio: all stdio operation could be done in userspace

## Testing

sim:nsh

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@michallenc this patch could make your patch(https://github.com/apache/nuttx/pull/10602) simpler.

File: sched/tls/task_uninitinfo.c [hunk 1]
```diff
@@ -24,11 +24,80 @@
...
+  /* Destroy the mutex and release the filelist */
+
+  nxmutex_destroy(&list->sl_lock);
+
+  /* Destroy stdin, stdout and stderr stream */
+
+#ifndef CONFIG_STDIO_DISABLE_BUFFERING
+  nxrmutex_destroy(&stream->fs_lock);
+  nxrmutex_destroy(&stream->fs_lock);
+  nxrmutex_destroy(&stream->fs_lock);
```
Comment (davids5) [inline sched/tls/task_uninitinfo.c:64]: ?

Comment (xiaoxiang781216) [reply sched/tls/task_uninitinfo.c:64]: @davids5 what's problem do you want to ask?

Comment (acassis) [reply sched/tls/task_uninitinfo.c:64]: @xiaoxiang781216 I think this question is same as mine: why  to destroy the same stream->fs_lock three times? Is it related to STDIN/STDOUT/STDERR ?

Comment (xiaoxiang781216) [reply sched/tls/task_uninitinfo.c]: Ok, done.

**Review by anchao — APPROVED** [review]
LGTM

**Pull request merged**